security-analysis-ws/src/main/java/eu/itesla_project/security/rest/api/impl/SecurityAnalysisServiceImpl.java

```diff
@@ -179,4 +183,140 @@ private FilePart getFilePart(List<InputPart> parts) throws IOException {
+            }
+
+            FilePart dslFile = formParts.get("dsl-file") != null ? getFilePart(formParts.get("dsl-file")) : null;
+            if (dslFile == null) {
+                return Response.status(Status.BAD_REQUEST).entity("Missing required dsl-file parameter").build();
+            }
+
+            Network network = Importers.loadNetwork(caseFile.getFilename(), caseFile.getInputStream());
+
+            boolean verbose = false;
+
+            List<String> contingencies = Collections.emptyList();
+
+            StringWriter writer = new StringWriter();
+            IOUtils.copy(dslFile.getInputStream(), writer, StandardCharsets.UTF_8);
+            String dslAsString = writer.toString();
+
+            // load actions from Groovy DSL
+            ActionDb actionDb = new ActionDslLoader(dslAsString).load(network,
+                    new ActionDslLoaderObserver(new PrintStream(new ByteArrayOutputStream()), verbose));
+
+            if (contingencies.isEmpty()) {
+                contingencies = actionDb.getContingencies().stream().map(Contingency::getId)
+                        .collect(Collectors.toList());
+            }
+
+            LoadFlowActionSimulatorConfig config = LoadFlowActionSimulatorConfig.load();
+
+            List<LoadFlowActionSimulatorObserver> observers = new ArrayList<>();
+            AbstractSecurityAnalysisResultBuilderImpl loadFlowActionSimulatorObserver = new AbstractSecurityAnalysisResultBuilderImpl();
+            observers.add(loadFlowActionSimulatorObserver);
+
+            // action simulator
+            ActionSimulator actionSimulator = new LoadFlowActionSimulator(network, new LocalComputationManager(), config,
+                    observers);
+
+            // start simulator
+            actionSimulator.start(actionDb, contingencies);
+
+            SecurityAnalysisResult securityAnalysisResult = loadFlowActionSimulatorObserver.getResult();
+
+            return Response.ok(toStream(securityAnalysisResult, network, format))
+                    .header("Content-Type", format.equals(Format.JSON) ? MediaType.APPLICATION_JSON : "text/csv")
+                    .build();
+
+        } catch (Exception e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Internal server error").build();
+        }
+    }
+
+    private static class AbstractSecurityAnalysisResultBuilderImpl extends AbstractSecurityAnalysisResultBuilder {
+
+        private SecurityAnalysisResult result;
+
+        @Override
+        public void onFinalStateResult(SecurityAnalysisResult result) {
+            this.setResult(result);
+        }
+
+        public SecurityAnalysisResult getResult() {
+            return result;
+        }
+
+        public void setResult(SecurityAnalysisResult result) {
+            this.result = result;
+        }
+
+    }
+
+    private static class ActionDslLoaderObserver extends DefaultActionDslLoaderObserver {
```

Comment:
See comment above: I think it's not useful to keep that class.
(Or if we keep it, it should log to a logger, with a more specific name than ActionDslLoaderObserver)